### PR TITLE
Decrease usage of typed clients in favour of controller-runtime client

### PR DESCRIPTION
**Pull request by fiunchinho**

### CODE REVIEW

**Review by tuommaki — APPROVED** [review]
The change itself looks fine, but not sure if it matches branch name/PR title?

**Comment by fiunchinho** [comment]
> The change itself looks fine, but not sure if it matches branch name/PR title?

True. Title/branch are too optimistic. These changes help going in that direction but we still have places where they are used.

**Comment by fiunchinho** [comment]
I pushed more changes to cover some other places where the typed clients were used.

File: service/controller/resource/nodepool/create_drain_old_worker_nodes.go [hunk 1]
```diff
@@ -57,20 +58,19 @@ func (r *Resource) drainOldWorkerNodesTransition(ctx context.Context, obj interf
 	r.Logger.LogCtx(ctx, "level", "debug", "message", "finding all drainerconfigs")
 
 	drainerConfigs := make(map[string]corev1alpha1.DrainerConfig)
+	var labelSelector client.MatchingLabels
```
Comment (tuommaki) [inline service/controller/resource/nodepool/create_drain_old_worker_nodes.go:61]: Could this be just:
```go
var labelSelector := client.MatchingLabels{label.Cluster: key.ClusterID(&azureMachinePool)}
```

**Review by tuommaki — APPROVED** [review]

File: service/controller/resource/nodepool/create_drain_old_worker_nodes.go (hunk 1 above)
Comment (fiunchinho) [reply service/controller/resource/nodepool/create_drain_old_worker_nodes.go]: it could!

**Pull request merged**